Sources/NIOCore/SocketAddresses.swift

```diff
@@ -445,6 +456,7 @@ public enum SocketAddress: CustomStringConvertible, Sendable {
         self = .v6(.init(address: ipv6Addr, host: ""))
     }
 
+#if !os(WASI)
```

Comment:
Could you make `makeAddressResolvingHost` just throw exception instead of guarding out the API itself?

Reply:
I'm unsure why we'd prefer runtime checks as opposed to compile-time checks, but this particular change seems easy, so I pushed it for now.